src/Text.Analyzers/Core/CodeAnalysisDictionary.cs

```diff
@@ -0,0 +1,134 @@
+        /// </summary>
+        /// <param name="other">The other instance whose values we should copy.</param>
+        protected CodeAnalysisDictionary(CodeAnalysisDictionary other)
+        {
+            RecognizedWords = new HashSet<string>(other.RecognizedWords, StringComparer.OrdinalIgnoreCase);
+            UnrecognizedWords = new HashSet<string>(other.UnrecognizedWords, StringComparer.OrdinalIgnoreCase);
+        }
+
+        /// <summary>
+        /// A list of misspelled words that the spell checker will now ignore.
+        /// </summary>
+        /// <example>
+        /// <code>
+        /// <Recognized>
+        ///     <Word>knokker</Word>
+        /// </Recognized>
+        /// </code>
+        /// </example>
+        public HashSet<string> RecognizedWords { get; protected set; } = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
+
+        /// <summary>
+        /// A list of correctly spelled words that the spell checker will now report.
+        /// </summary>
+        /// <example>
+        /// <code>
+        /// <Unrecognized>
+        ///     <Word>meth</Word>
+        /// </Unrecognized>
+        /// </code>
+        /// </example>
+        public HashSet<string> UnrecognizedWords { get; protected set; } = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
+
+        /// <summary>
+        /// Creates a new instance of this class with recognized and unrecognized words (if specified) loaded
+        /// from the specified XML <paramref name="streamReader"/>.
+        /// </summary>
+        /// <param name="streamReader">XML stream of a code analysis dictionary.</param>
+        /// <returns>A new instance of this class with the words loaded.</returns>
+        public static CodeAnalysisDictionary CreateFromXml(StreamReader streamReader)
+        {
+            var document = XDocument.Load(streamReader);
+            return new CodeAnalysisDictionary(document);
+        }
+
+        /// <summary>
+        /// Creates a new instance of this class with recognized words loaded from the specified DIC <paramref name="streamReader"/>.
+        /// </summary>
+        /// <param name="streamReader">DIC stream of a code analysis dictionary.</param>
+        /// <returns>A new instance of this class with recognized words loaded.</returns>
+        public static CodeAnalysisDictionary CreateFromDic(StreamReader streamReader)
+        {
+            var words = new Collection<string>();
+
+            string word;
+            while ((word = streamReader.ReadLine()) != null)
+            {
+                if (!string.IsNullOrWhiteSpace(word.Trim()))
+                {
+                    words.Add(word);
+                }
+            }
+
+            return new CodeAnalysisDictionary(words);
+        }
+
+        /// <summary>
+        /// Returns a copy of the current instance.
+        /// </summary>
+        /// <returns></returns>
+        public CodeAnalysisDictionary Clone() => new CodeAnalysisDictionary(this);
```

Comment:
💭 If we make `CodeAnalysisDictionary` immutable, this is no longer required

Reply:
Done